compiler/compile/VirtualGuard.cpp

```diff
@@ -227,38 +227,31 @@ TR_VirtualGuard::createBreakpointGuardNode
 #ifdef J9_PROJECT_SPECIFIC
 /*
  * Shape of a breakpoint guard:
- * iflcmpeq/ificmpeq
- *    land/iand
- *       lloadi/iloadi <ConstantPool shadow>
- *          aconst J9Method
+ * ificmpne
+ *    iand
+ *       bu2i
+ *          bloadi <ConstantPool shadow>
  *       isBreakpointedBit 
- *    isBreakpointedBit 
+ *    iconst 0
  */
    bool is64Bit = TR::Compiler->target.is64Bit();
    TR::SymbolReferenceTable * symRefTab = comp->getSymRefTab();
    TR::SymbolReference * fieldSymRef = symRefTab->findOrCreateJ9MethodConstantPoolFieldSymbolRef(offsetof(struct J9Method, constantPool));
    TR::Node * aconstNode = TR::Node::aconst(callNode, (uintptrj_t)calleeSymbol->getResolvedMethod()->getPersistentIdentifier());
-   TR::Node * constantPool = TR::Node::createWithSymRef(is64Bit? TR::lloadi: TR::iloadi, 1, 1, aconstNode, fieldSymRef);
    aconstNode->setIsMethodPointerConstant(true);
    aconstNode->setInlinedSiteIndex(calleeIndex);
    aconstNode->setByteCodeIndex(0);
    TR::Node * flagBit = NULL;
    TR::Node *guard = NULL;
-   if (TR::Compiler->target.is64Bit())
-      {
-      flagBit = TR::Node::create(callNode, TR::lconst, 0, 0);
-      flagBit->setLongInt(comp->fej9()->offsetOfMethodIsBreakpointedBit());
 
```

Comment:
So this is a 32 bit field I assume regardless of whether one is on 64-bit or 32-bit platform ?

Reply:
Thanks for the comments! The intention was to only load the last byte of that field to make the instruction sequence faster but on a second thought, this won't work for big endian platforms. I changed it back to the old implementation. The efficiency of the generated sequence shouldn't matter anyway because we would nop them.